src/utils.js

```diff
@@ -86,18 +86,42 @@ export async function asyncBufferFromUrl({ url, byteLength, requestInit }) {
   if (!url) throw new Error('missing url')
   // byte length from HEAD request
   byteLength ||= await byteLengthFromUrl(url, requestInit)
+
+  /**
+   * A promise for the whole buffer, if range requests are not supported.
+   * @type {Promise<ArrayBuffer>|undefined}
+   */
+  let buffer = undefined
   const init = requestInit || {}
+
   return {
     byteLength,
     async slice(start, end) {
-      // fetch byte range from url
+      if (buffer) {
+        return buffer.then(buffer => buffer.slice(start, end));
+      }
+
       const headers = new Headers(init.headers)
       const endStr = end === undefined ? '' : end - 1
       headers.set('Range', `bytes=${start}-${endStr}`)
+
       const res = await fetch(url, { ...init, headers })
       if (!res.ok || !res.body) throw new Error(`fetch failed ${res.status}`)
-      return res.arrayBuffer()
-    },
+      
+      switch (res.status) {
+        // Endpoint does not support range requests and returned the whole object
+        case 200:
+          buffer = res.arrayBuffer();
```

Comment:
lint: should be no semi-colons at end of line. Please run `npm run lint` and fix the errors.

(I'm not sure why lint didn't run on this PR? It should be in the github actions?)

Reply:
It looks like the CI requires approval to run: https://github.com/hyparam/hyparquet/actions/runs/12801142297

> This workflow is awaiting approval from a maintainer in https://github.com/hyparam/hyparquet/pull/57

Maybe a repo setting to change

Reply:
I just launched the CI